### PR TITLE
Adsk contrib/image address mode updates

### PR DESCRIPTION
**Pull request by bernardkwok**
- Add in mirror and default address mode into enumerated options for <image>
- Update GLSL texture address mode binding to support new modes. (No change to OSL as names match possible values for uwrap and vwrap).
- Update unit tests.

### CODE REVIEW

File: libraries/stdlib/stdlib_defs.mtlx [hunk 1]
```diff
@@ -69,20 +69,20 @@
...
     <input name="texcoord" type="vector2" defaultgeomprop="UV0" uiname="Texture Coordinates"/>
-    <parameter name="uaddressmode" type="string" value="periodic" enum="black,clamp,periodic" uiname="Address Mode U"/>
-    <parameter name="vaddressmode" type="string" value="periodic" enum="black,clamp,periodic" uiname="Address Mode V"/>
+    <parameter name="uaddressmode" type="string" value="periodic" enum="black,clamp,periodic,mirror,default" uiname="Address Mode U"/>
+    <parameter name="vaddressmode" type="string" value="periodic" enum="black,clamp,periodic,mirror,default" uiname="Address Mode V"/>
     <parameter name="filtertype" type="string" value="linear" enum="closest,linear,cubic" uiname="Filter Type"/>
     <parameter name="framerange" type="string" value="" uiname="Frame Range"/>
     <parameter name="frameoffset" type="integer" value="0" uiname="Frame Offset"/>
-    <parameter name="frameendaction" type="string" value="black" enum="black,clamp,periodic" uiname="Frame End Action"/>
+    <parameter name="frameendaction" type="string" value="black" enum="black,clamp,periodic,default" uiname="Frame End Action"/>
```
Comment (niklasharrysson) [inline libraries/stdlib/stdlib_defs.mtlx]: Was this intentional? If so I guess all versions should have it?

File: source/MaterialXRenderGlsl/GLTextureHandler.cpp [hunk 2]
```diff
@@ -210,18 +214,30 @@ bool GLTextureHandler::bindImage(const FilePath& filePath, const ImageSamplingPr
...
+        // Mirror
+        GL_MIRRORED_REPEAT,
+
+        // Mapping is from "default". Use clamp to border
+        // with default color to achieve this
+        GL_CLAMP_TO_BORDER
+    };
+
+    int addressMode = GL_REPEAT;
+    if (addressModeEnum >= 0 && addressModeEnum < 5)
```
Comment (niklasharrysson) [inline source/MaterialXRenderGlsl/GLTextureHandler.cpp]: Maybe the MaterialX adressmodes should be enums to increase code readability.

File: libraries/stdlib/stdlib_defs.mtlx (hunk 1 above)
Comment (bernardkwok) [reply libraries/stdlib/stdlib_defs.mtlx]: Good catch. String replacement error. I'll fix this up.

File: source/MaterialXRenderGlsl/GLTextureHandler.cpp (hunk 2 above)
Comment (bernardkwok) [reply source/MaterialXRenderGlsl/GLTextureHandler.cpp]: Yes. There is a bit of "magic" numbering here. I'll fix.

File: source/MaterialXRenderGlsl/GLTextureHandler.cpp [hunk 3]
```diff
@@ -208,34 +208,40 @@ bool GLTextureHandler::bindImage(const FilePath& filePath, const ImageSamplingPr
...
 
-int GLTextureHandler::mapAddressModeToGL(int addressModeEnum)
+int GLTextureHandler::mapAddressModeToGL(ImageSamplingProperties::AddressMode addressModeEnum)
 {
-    int addressMode = GL_REPEAT;
-
-    // Mapping is from "black". Use clamp to border
-    // with border color black to achieve this
-    if (addressModeEnum == 0)
+    const vector<int> addressModes =
```
Comment (feldstj) [inline source/MaterialXRenderGlsl/GLTextureHandler.cpp]: I don't think you need the = sign here

**Pull request merged**